Release Godot 4.2 dev1 and publish release notes (#680)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-2-dev-1.md

---
title: "Dev snapshot: Godot 4.2 dev 1"
excerpt: "We are ready to share the first batch of improvements coming to Godot 4.2 later this year with the release of the first dev snapshot!"
categories: ["pre-release"]
author: Yuri Sizov
image: /storage/blog/covers/dev-snapshot-godot-4-2-dev-1.webp
image_caption_title: "Terrain3D"
image_caption_description: "A tool for 3D terrain creation by Outobugi Games and Tokisan Games"
date: 2023-07-19 15:00:00
---

> Two weeks ago we wrapped up the development of [Godot 4.1](/article/godot-4-1-is-here), bringing more polish and stability to the Godot 4 series, and just two days ago we shipped its [first maintenance release](/article/maintenance-release-godot-4-1-1), addressing some of the more urgent issues that slipped in. In preparation to both of these releases many improvements were temporarily put aside to ensure stability and lack of last minute regressions.

Reviewer: ```suggestion
Two weeks ago we wrapped up the development of [Godot 4.1](/article/godot-4-1-is-here), bringing more polish and stability to the Godot 4 series, and just two days ago we shipped its [first maintenance release](/article/maintenance-release-godot-4-1-1), addressing some of the more urgent issues that slipped in. In preparation for both of these releases we temporarily put aside many improvements to ensure stability and to avoid any last minute regressions.
```

> Today, we are ready to share these improvements with you with **4.2 dev 1** being available for public testing (read about most notable changes [below](#whats-new)). This first development snapshot marks the proper start of the new development cycle, which means in the coming weeks the engine will be changed aggressively as contributors implement new features and fix long-standing bugs. You should expect a dev snapshot every 1 or 2 weeks, depending on how stable the development branch is after any given series of merges.

Reviewer: ```suggestion
Today, we are ready to share these improvements with you with **4.2 dev 1** being available for public testing (you can read about the most notable changes [below](#whats-new)). This first development snapshot marks the proper start of the new development cycle, which means in the coming weeks we will be quickly merging many changes as contributors implement new features and fix long-standing bugs. You should expect a dev snapshot every 1 or 2 weeks, depending on how stable the development branch is after any given series of merges.

Please note that dev1 includes many changes that just barely missed being included in 4.1, so expect it to be a little bit more feature packed than normal dev builds. 
```

- Particles: Add `finished` signal to CPUParticles and GPUParticles ([GH-76853](https://github.com/godotengine/godot/pull/76853), [GH-76859](https://github.com/godotengine/godot/pull/76859)).
- Physics: Add `hit_back_faces` property to `RayCast3D` ([GH-79330](https://github.com/godotengine/godot/pull/79330)).
- Porting: Implement native file selection dialog support ([GH-47499](https://github.com/godotengine/godot/pull/47499), [GH-79574](https://github.com/godotengine/godot/pull/79574)).
  - This is only implemented for macOS and Windows so far. The macOS implementation requires the app to be sandboxed.
- Porting: Android: Refactor Godot Android architecture ([GH-76821](https://github.com/godotengine/godot/pull/76821)).
- Porting: macOS: Fix uncapped frame rate for windows in the non-active workspaces ([GH-79572](https://github.com/godotengine/godot/pull/79572)).
- Rendering: Draw frustum splices on top of direction shadow atlas for debug purposes ([GH-77085](https://github.com/godotengine/godot/pull/77085)).
- Rendering: Split raster barrier into vertex and fragment barrier ([GH-77420](https://github.com/godotengine/godot/pull/77420)).

>   - This change currently breaks compatibility by modifying some of the values for the `BarrierMask` enumeration.

Reviewer: ```suggestion
  - This change currently breaks compatibility by modifying some of the values for the `BarrierMask` enumeration. If you don't rely on the values of `BarrierMask`, no modifications are necessary.
```

- Multiplayer: Use `get/set_indexed` in MultiplayerSynchronizer ([GH-79479](https://github.com/godotengine/godot/pull/79479)).
  - This change allows synchronizing (sub-)resource properties, transform components, etc. without having to synchronize the entire object.
- Navigation: Add NavigationRegion function to change navigation map ([GH-77191](https://github.com/godotengine/godot/pull/77191)).
- Navigation: Mark `NavigationServer3D.region_bake_navigation_mesh` as deprecated ([GH-79137](https://github.com/godotengine/godot/pull/79137)).
- Navigation: Change 2D navigation ProjectSettings from integers to floats ([GH-79483](https://github.com/godotengine/godot/pull/79483)).
- Particles: Add `finished` signal to CPUParticles and GPUParticles ([GH-76853](https://github.com/godotengine/godot/pull/76853), [GH-76859](https://github.com/godotengine/godot/pull/76859)).
- Physics: Add `hit_back_faces` property to `RayCast3D` ([GH-79330](https://github.com/godotengine/godot/pull/79330)).
- Porting: Implement native file selection dialog support ([GH-47499](https://github.com/godotengine/godot/pull/47499), [GH-79574](https://github.com/godotengine/godot/pull/79574)).

>   - This is only implemented for macOS and Windows so far. The macOS implementation requires the app to be sandboxed.

Reviewer 2: > The macOS implementation requires the app to be sandboxed.

It's not. Native dialog is required for a sandboxed app to access files outside the sandbox. But it can be used in any app.

Author: Oh, I misunderstood the title of your PR then. I'll fix the blogpost
